simulator/examples/hello.rs

```diff
@@ -1,3 +1,7 @@
+//! # Example: Hello
```

Comment:
I was thinking about eventually removing this example entirely. IMO this doesn't provide any additional value over the new `hello-world` example. Thoughts?

Reply:
Yeah it's pretty garbage! I've removed it, and migrated the `chaining.rs` example to mirror the `hello-world.rs` output.